pyterrier_rag/_optional.py

```diff
@@ -1,27 +1,46 @@
+from importlib.util import find_spec
+
+
 def is_openai_availible():
     try:
-        import openai # noqa: F401
+        find_spec("openai")
+
         return True
-    except ImportError:
+    except ValueError:
         return False
 
+
+def is_transformers_availible():
+    try:
+        find_spec("transformers")
```

Comment:
I don't think this works as expected. It looks like `find_spec` returns `None` if not found, rather than raising a `ValueError`. https://docs.python.org/3/library/importlib.html#importlib.abc.MetaPathFinder.find_spec